go/vt/vtgate/planbuilder/insert.go

```diff
@@ -232,49 +355,68 @@ func generateInsertShardedQuery(node *sqlparser.Insert, eins *engine.Insert, val
 	eins.Suffix = suffixBuf.String()
 }
 
-// modifyForAutoinc modfies the AST and the plan to generate
-// necessary autoinc values. It must be called only if eins.Table.AutoIncrement
-// is set. Bind variable names are generated using baseName.
-func modifyForAutoinc(ins *sqlparser.Insert, eins *engine.Insert) error {
-	colNum := findOrAddColumn(ins, eins.Table.AutoIncrement.Column)
-	rows := ins.Rows.(sqlparser.Values)
-	autoIncValues := make([]evalengine.Expr, 0, len(rows))
-	for rowNum, row := range rows {
-		// Support the DEFAULT keyword by treating it as null
-		if _, ok := row[colNum].(*sqlparser.Default); ok {
-			row[colNum] = &sqlparser.NullVal{}
-		}
+func generateInsertSelectQuery(node *sqlparser.Insert, eins *engine.Insert) {
+	prefixBuf := sqlparser.NewTrackedBuffer(dmlFormatter)
+	suffixBuf := sqlparser.NewTrackedBuffer(dmlFormatter)
+	prefixBuf.Myprintf("insert %v%sinto %v%v ",
+		node.Comments, node.Ignore.ToString(),
+		node.Table, node.Columns)
```

Comment:
Do we need a space between the comments and the ignore, or `node.Comments` already contain a trailing space?

Reply:
parser works fine with comments without space. if comments does not exists then the query will have 2 space.